db/schema_applier.cc

```diff
@@ -845,11 +850,34 @@ future<> schema_applier::notify() {
         for (auto& name : _affected_keyspaces.dropped) {
             co_await notifier.drop_keyspace(name);
         }
+        // notify about user types
+        auto& types = _affected_user_types.per_shard[this_shard_id()];
+        for (auto& type : types.created) {
+            co_await notifier.create_user_type(type);
+        }
+        for (auto& type : types.altered) {
+            co_await notifier.update_user_type(type);
+        }
+        for (auto& type : types.dropped) {
+            co_await notifier.drop_user_type(type);
+        }
     });
```

Comment:
Can the notification reordering affect prepared statement invalidation? A prepared statement can refer to both tables and user types.

Reply:
good question, we have some dtests for notifications but quite minimal, I'll consult someone from drivers or ask maybe they have more detailed tests

I think per shard there should be no re-ordering of notifications introduced (at PR's head), but there might be across shards.